File: 292_can_win_nim/can_win_nim.cpp
/*
 * @Date: 2021-09-18 08:41:38
 * @Author: Mengsen Wang
 * @LastEditors: Mengsen Wang
 * @LastEditTime: 2021-09-18 08:42:40
 */

#include <cassert>

class Solution {
 public:
  bool canWinNim(int n) { return n % 4 != 0; }
};

int main() {
  assert(!Solution().canWinNim(4));
  assert(Solution().canWinNim(1));
  assert(Solution().canWinNim(2));
}

File: 292_can_win_nim/can_win_nim.rs
/*
 * @Date: 2021-09-18 08:42:58
 * @Author: Mengsen Wang
 * @LastEditors: Mengsen Wang
 * @LastEditTime: 2021-09-18 08:49:16
 */

struct Solution;

impl Solution {
    pub fn can_win_nim(n: i32) -> bool {
        n % 4 != 0
    }
}

fn main() {
    assert!(!Solution::can_win_nim(4));
    assert!(Solution::can_win_nim(1));
    assert!(Solution::can_win_nim(2));
}
